pgrx-sql-entity-graph/src/used_type.rs

```diff
@@ -608,6 +613,143 @@ fn resolve_option_inner(
+                } else if path_arg.args.len() == 2 {
+                    (first_ty.clone(), Some(path_arg.args[1].clone()))
+                } else {
+                    return Err(syn::Error::new(
+                        last.arguments.span(),
+                        "Result<..> with more than two type arguments not supported.",
+                    ));
+                }
+            } else {
+                // Return early, Result<> with no type args.
+                return Err(syn::Error::new(
+                    last.arguments.span(),
+                    "Cannot return a Result without type generic arguments.",
+                ));
+            }
+        } else {
+            // Return early, invalid signature for Result<T,E>
+            return Err(syn::Error::new(
+                last.arguments.span(),
+                "Cannot return a Result without type generic arguments.",
+            ));
+        }
+    };
+
+    // Inner / nested function for getting a type signature for a Result from
+    // the tuple of (ok_type, Option<error_type>)
+    fn type_for_args(
+        no_args_path: syn::Path,
+        first_ty: syn::Type,
+        err_ty: Option<GenericArgument>,
+    ) -> syn::Type {
+        match err_ty {
+            Some(e) => {
+                syn::parse_quote! {
+                    #no_args_path<#first_ty, #e>
+                }
+            }
+            None => {
+                // Since `pub type Result<T> = std::error::Result<T, OurError>
+                // is a common pattern,
+                // we should support single-argument Result<T> style.
+                syn::parse_quote! {
+                    #no_args_path<#first_ty>
+                }
+            }
+        }
+    }
+
+    match &ok_ty {
+        syn::GenericArgument::Type(ty) => {
+            match ty.clone() {
+                syn::Type::Macro(macro_pat) => {
+                    let mac = &macro_pat.mac;
+                    let archetype = mac.path.segments.last().expect("No last segment");
+                    match archetype.ident.to_string().as_str() {
+                        // Result<composite_type!(..), E>
+                        "composite_type" => {
+                            let composite_mac = handle_composite_type_macro(mac)?;
+                            let comp_ty = composite_mac.expand_with_lifetime();
+                            let sql = Some(composite_mac);
+
+                            let ty = type_for_args(without_type_args, comp_ty, err_ty);
+                            Ok((ty, sql))
+                        },
+                        // Option<default!(composite_type!(..))> isn't valid. If the user wanted the default to be `NULL` they just don't need a default.
+                        "default" => Err(syn::Error::new(mac.span(), "`Result<default!(T, \"my_default\"), E>` not supported, choose `Result<T, E>` for a default of `NULL`, or `default!(T, default)` for a non-NULL default")),
```

Comment:
For the SQL NULL default case, they want `Result<Option<T>, E>`, right?

Reply:
Honestly this is just something I forgot to change. I have fixed it now.